src/Uno.UI/UI/Xaml/Media/Imaging/ImageSourceHelpers.cs

```diff
@@ -34,16 +35,13 @@ public static async Task<ImageData> ReadFromStreamAsBytesAsync(Stream stream, Ca
 #if __SKIA__
 	public static Task<ImageData> ReadFromStreamAsCompositionSurface(Stream imageStream, CancellationToken ct)
 	{
-		var surface = new Microsoft.UI.Composition.SkiaCompositionSurface();
-		var result = surface.LoadFromStream(imageStream);
-
-		if (result.success)
+		if (SkiaCompositionSurface.LoadFromStream(imageStream) is { } surface)
 		{
 			return Task.FromResult(ImageData.FromCompositionSurface(surface));
 		}
 		else
 		{
-			var exception = new InvalidOperationException($"Image load failed ({result.nativeResult})");
+			var exception = new InvalidOperationException($"Image load failed");
```

Comment:
There are some case where the `SkiaCompositionSurface.LoadFromStream` raise an exception that is logged, dropped and we return `null`. We should not hide useful information to the users on the reason why the image load failed. You should instead changed the `LoadFromStream` (or create an overload):

```
bool SkiaCompositionSurface.TryLoadFromStream(
    [NotNullWhen(true)] SkiaCompositionSurface? surface, 
    [NutNullWhen(false) Exception? error)
```

and actually thinking about it, as the `SkiaCompositionSurface` only contains an `SKImage` we could even drop it  and just return an `SKImage`:


```
bool SkiaCompositionSurface.TryLoadFromStream(
    [NotNullWhen(true)] SKImage? image, 
    [NutNullWhen(false) Exception? error)
```

Not sure if we could image scenarios where we would return something else than an `SKImage` ?

Reply:
Love this pattern. As for returning an SKImage directly: SkiaCompositionSurface will soon be more complex than a simple SKImage (https://github.com/unoplatform/uno/pull/15951/).